packages/frontend-defaults/src/createApp.tsx

```diff
@@ -87,29 +94,93 @@ export function createApp(options?: CreateAppOptions): {
         overrideBaseUrlConfigs(defaultConfigLoaderSync()),
       );
 
-    const discoveredFeatures = getAvailableFeatures(config);
+    const loadedFeatures: FrontendFeature[] = [];
+
+    // Separate deprecated CreateAppFeatureLoader elements from the frontend features,
+    // and manage the deprecated elements first.
+    const [providedFeatures, createAppFeatureLoaders] = (
+      options?.features ?? []
+    ).reduce<[FrontendFeature[], CreateAppFeatureLoader[]]>(
+      (acc, item) => {
+        if ('$$type' in item) {
+          acc[0].push(item);
+        } else {
+          acc[1].push(item);
+        }
+        return acc;
+      },
+      [[], []],
+    );
+    for (const entry of createAppFeatureLoaders) {
+      try {
+        const result = await entry.load({ config });
+        providedFeatures.push(...result.features);
+      } catch (e) {
+        throw new Error(
+          `Failed to read frontend features from loader '${entry.getLoaderName()}', ${stringifyError(
+            e,
+          )}`,
+        );
+      }
+    }
 
-    const providedFeatures: FrontendFeature[] = [];
-    for (const entry of options?.features ?? []) {
-      if ('load' in entry) {
+    const featureLoaderNames: InternalFrontendFeatureLoader[] = [];
+    const maxRecursionDepth = options?.featureLoaderRecursionDepth ?? 10;
+
+    async function applyFeatureLoaders(
+      features: FrontendFeature[],
+      recursionDepth: number,
+    ) {
+      if (features.length === 0) {
+        return;
+      }
+      const [featureLoaders, otherFeatures] = features.reduce<
+        [InternalFrontendFeatureLoader[], FrontendFeature[]]
+      >(
+        (acc, item) => {
+          if (isInternalFrontendFeatureLoader(item)) {
+            acc[0].push(item);
+          } else {
+            acc[1].push(item);
+          }
+          return acc;
+        },
+        [[], []],
+      );
+      loadedFeatures.push(...otherFeatures);
+      for (const featureLoader of featureLoaders) {
+        if (featureLoaderNames.some(l => l === featureLoader)) {
+          throw new Error(
+            `Added several instances of feature loader ${featureLoader.description}`,
+          );
```

Comment:
I think a `continue` here would be fine tbh

Reply:
What would then be the option you would recommend to at least log a warning ?